cpp-examples/mutability.cpp

```diff
@@ -1,31 +1,32 @@
 #include <vector>
 #include <iostream>
 
-void print_shopping_list(std::vector<std::string>* shopping_list) {
-    if (shopping_list->size() == 0) {
-        std::cout << "\t* nothing\n";
+void print_shopping_list(std::vector<std::string> const &shopping_list) {
+    //                                            ^^^^^ explicitly immmutable
+    if (shopping_list.size() == 0) {
+        std::cout << "\t-- nothing\n";
     }
 
-    for (auto item = shopping_list->begin(); item != shopping_list->end(); ++item) {
-        std::cout << "\t* " << *item << '\n';
+    for (std::string item: shopping_list) {
```

Comment:
this could be `for (const auto& item : shopping_list)`. Sorry, that I am a little bit nitpicky.

Reply:
please be, this is your turf!
fixed.